impl/src/main/java/io/jsonwebtoken/impl/DefaultJwtParser.java

```diff
@@ -317,6 +327,9 @@ public Jwt parse(String jwt) throws ExpiredJwtException, MalformedJwtException,
         }
 
         // =============== Body =================
+        if (base64UrlEncodedPayload == null) {
```

Comment:
if base64UrlEncodedPayload is `null`, we don't need to execute lines 333-337 below (no need to incur overhead of a Base64 Decoder and a CompressionCodec if there is nothing to decode or decompress).

Based on that, maybe the logic should look something like this:

```java
String payload = "";
if (base64UrlEncodedPayload != null) {
    byte[] bytes = base64UrlDecoder.decode(base64UrlEncodedPayload);
    if (compressionCodec != null) {
        bytes = compressionCodec.decompress(bytes);
    }
    payload = new String(bytes, Strings.UTF_8);
}
```

Reply:
Hi!
alright, I removed the `payloadRequired` methods and logic.

One last thing: There were each three tests in `JwtsTest` and `DeprecatedJwtsTest` regarding missing empty payloads:
* `testParseWithHeaderOnly` is now removed because given header `foo` is no valid base64 encoded header.
* `testParseWithHeaderAndSignatureOnly` is now removed because given header `foo` is no valid base64 encoded header and `bar` is no valid signature.
* `testParseWithTwoPeriodsOnly` is **ignored** for now, because while reading through the RFC I was not sure whether `..` is a valid JWT or not. What should we do about this?

Otherwise empty payloads are now supported. Thanks for your support. :-)

Reply:
(Unresolved conversation because of open question)